apps/api/v1/lib/utils/isAdmin.ts

```diff
@@ -33,7 +33,27 @@ export const isAdminGuard = async (req: NextApiRequest) => {
       },
     },
   });
-  if (orgOwnerOrAdminMemberships.length > 0) return { isAdmin: true, scope: ScopeOfAdmin.OrgOwnerOrAdmin };
+
+  if (orgOwnerOrAdminMemberships.length > 0) {
+    const permissionCheckService = new PermissionCheckService();
+
+    // Check PBAC permissions for each organization
+    // This should only ever be one membership as we only support one org currently.
+    for (const membership of orgOwnerOrAdminMemberships) {
+      const teamId = membership.team.id;
+
+      const hasAdminPermission = await permissionCheckService.checkPermission({
+        userId,
+        teamId,
+        permission: "organization.adminApi",
+        fallbackRoles: [MembershipRole.OWNER, MembershipRole.ADMIN],
+      });
```

Comment:
Check if this user has the org adminApi permission if PBAC is enabled. OR fall back to checking if they are a owner or admin of this team